samtranslator/swagger/swagger.py

```diff
@@ -788,6 +788,50 @@ def add_models(self, models):
 
             self.definitions[model_name.lower()] = schema
 
+    def add_request_parameters_to_method(self, path, method_name, request_parameters):
+        """
+        Add Parameters to Swagger.
+
+        :param string path: Path name
+        :param string method_name: Method name
+        :param list request_parameters: Dictionary of Parameters
+        :return:
+        """
+
+        normalized_method_name = self._normalize_method_name(method_name)
+        # It is possible that the method could have two definitions in a Fn::If block.
+        for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
+
+            # If no integration given, then we don't need to process this definition (could be AWS::NoValue)
+            if not self.method_definition_has_integration(method_definition):
+                continue
+
+            existing_parameters = method_definition.get('parameters', [])
+
+            for request_parameter in request_parameters:
+
+                parameter_name = request_parameter['Name']
+                location_name = parameter_name.replace('method.request.', '')
+                location, name = location_name.split('.')
+
+                parameter = {
+                    'in': location,
```

Comment:
Ok here in the swagger doc, you do need to use `query` instead of `querystring`, but what I'm not clear on is if you need to use `querystring` in the API Gateway `cacheKeyParameters` section. Have you tested and verified this works correctly?

Reply:
It turns out the cacheKeyParameters requires `querystring` and not `query`.  Therefore I've updated the code and docs to require `querystring` and convert `querystring` to `query` for the swagger.